addons/binding/org.openhab.binding.yamahareceiver2/src/main/java/org/openhab/binding/yamahareceiver2/discovery/ZoneDiscoveryService.java

```diff
@@ -0,0 +1,61 @@
+package org.openhab.binding.yamahareceiver2.discovery;
+
+import static org.openhab.binding.yamahareceiver2.YamahaReceiverBindingConstants.THING_TYPE_YAMAHAAV;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.yamahareceiver2.YamahaReceiverBindingConstants;
+import org.openhab.binding.yamahareceiver2.internal.YamahaReceiverState;
+import org.openhab.binding.yamahareceiver2.internal.protocol.YamahaReceiverCommunication.Zone;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+
+public class ZoneDiscoveryService extends AbstractDiscoveryService {
```

Comment:
Somehow the zones do not really qualify to be modelled as a Thing on its own - they are rather a part of the Yamaha receiver. Do zones have exactly the same channels as the "main" zone? Doesn't the receiver have any further features like a "master volume" that controls all zones at once?
In any case, my suggestion would be to define a channel group for the existing channels and then you can have one of such channel groups per zone of your receiver. Then there would be no need for this additional discovery service and for additional things.